sync/syncer_test.go

```diff
@@ -807,3 +708,142 @@ func TestSyncer_Txs(t *testing.T) {
+
+	block4 := types.NewExistingBlock(types.BlockID(222), lid, nil)
+
+	syncObj1.AddBlockWithTxs(block4, []*types.AddressableSignedTransaction{tx4}, []*types.ActivationTx{atx4})
+
+	timeout := time.NewTimer(2 * time.Second)
+
+	wrk, output := NewPeersWorker(syncObj, []p2p.Peer{nodes[1].PublicKey()}, &sync.Once{}, LayerIdsReqFactory(lid))
+	go wrk.Work()
+
+	select {
+	case intr := <-output:
+		ids := intr.([]types.BlockID)
+		for _, a := range layer.Blocks() {
+			found := false
+			for _, id := range ids {
+				if a.ID() == types.BlockID(id) {
+					found = true
+					break
+				}
+			}
+			if !found {
+				t.Error(errors.New("id list did not match"))
+			}
+		}
+	case <-timeout.C:
+		assert.Fail(t, "no message received on channel")
+	}
+
+}
+
+func TestSyncProtocol_FetchBlocks(t *testing.T) {
+	syncs, nodes := SyncMockFactory(2, conf, "TestSyncProtocol_FetchBlocks_", memoryDB, newMockPoetDb)
+	atx1 := atx("trteyjuk")
+
+	atx2 := atx("he6gdsdf")
+	atx3 := atx("yturjasd")
+	syncObj1 := syncs[0]
+	defer syncObj1.Close()
+	syncObj2 := syncs[1]
+	defer syncObj2.Close()
+	pm1 := getPeersMock([]p2p.Peer{nodes[0].PublicKey()})
+	syncObj1.Log.Info("started fetch_blocks")
+	syncObj2.Peers = pm1 //override peers with
+
+	syncObj1.ProcessAtx(atx3)
+	block1 := types.NewExistingBlock(types.BlockID(123), 0, nil)
+	block1.ATXID = atx3.Id()
+	block2 := types.NewExistingBlock(types.BlockID(321), 1, nil)
+	block2.ATXID = atx3.Id()
+	block3 := types.NewExistingBlock(types.BlockID(222), 2, nil)
+	block3.ATXID = atx3.Id()
+
+	syncObj1.AddBlockWithTxs(block1, []*types.AddressableSignedTransaction{tx1, tx2, tx3, tx4, tx5, tx6, tx7, tx8}, []*types.ActivationTx{atx1})
+	syncObj1.AddBlockWithTxs(block2, []*types.AddressableSignedTransaction{tx1, tx2, tx3, tx4, tx5, tx6, tx7, tx8}, []*types.ActivationTx{atx2})
+	syncObj1.AddBlockWithTxs(block3, []*types.AddressableSignedTransaction{tx1, tx2, tx3, tx4, tx5, tx6, tx7, tx8}, []*types.ActivationTx{atx3})
+
+	ch := make(chan types.BlockID, 3)
+	ch <- block1.ID()
+	ch <- block2.ID()
+	ch <- block3.ID()
+
+	output := syncObj2.fetchWithFactory(NewBlockWorker(syncObj2, 1, BlockReqFactory(), blockSliceToChan([]types.BlockID{block1.ID(), block2.ID(), block3.ID()})))
+
+	for out := range output {
+		block := out.(*types.Block)
+		txs, err := syncObj2.syncTxs(block.TxIds)
+		if err != nil {
+			t.Error("could not fetch all txs", err)
+		}
+		atxs, err := syncObj2.syncAtxs(block.AtxIds)
+		if err != nil {
+			t.Error("could not fetch all atxs", err)
+		}
+		syncObj2.Debug("add block to layer %v", block)
+		syncObj2.AddBlockWithTxs(block, txs, atxs)
+	}
+}
+
+func TestSyncProtocol_syncAtxsWithAtxsInPoolNotInDb(t *testing.T) {
+	// test to verify the fix for bug #1128
+	syncs, nodes := SyncMockFactory(2, conf, "TestSyncProtocol_syncAtxs_", memoryDB, newMockPoetDb)
+	syncObj1 := syncs[0]
+	defer syncObj1.Close()
+	syncObj2 := syncs[1]
```

Comment:
can you do this for tx's as well  ? if not lets add it to the test plan

Reply:
Added. Good thing that you assisted on that, there was the same bug on the Txs code